cmd/spire-server/cli/validate/validate_posix_test.go

```diff
@@ -0,0 +1,259 @@
+//go:build !windows
+
+package validate
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	common_cli "github.com/spiffe/spire/pkg/common/cli"
+	"github.com/spiffe/spire/pkg/common/fflag"
+	"github.com/stretchr/testify/require"
+)
+
+func serverConfigWithPlugins(pluginConfig string) string {
+	return `
+server {
+	trust_domain = "example.org"
+  data_dir = "/opt/spire/server/data"
+}
+
+plugins {
+` + pluginConfig + `
+}
+`
+}
+
+func TestValidateConfig(t *testing.T) {
+	configDir := t.TempDir()
+
+	testCases := []struct {
+		name                   string
+		config                 string
+		expectedReturn         int
+		expectedErrors         []string
+		expectedStderr         string
+		pluginDataFilePath     string
+		pluginDataFileContents string
+	}{
+		{
+			name:           "empty config",
+			config:         serverConfigWithPlugins(""),
+			expectedReturn: 1,
+			expectedErrors: []string{
+				"'datastore' must be configured",
+				"KeyManager:\n\t\tconstraint not satisfied: expected exactly 1 but got 0\n",
+			},
+		},
+		{
+			name: "valid config",
+			config: serverConfigWithPlugins(`
+DataStore "sql" {
+    plugin_data {
+        database_type = "sqlite3"
+        connection_string = "/some/path/to/the/datastore.sqlite3"
+    }
+}
+KeyManager "memory" {
+    plugin_data = {}
+}
+
+UpstreamAuthority "disk" {
+	plugin_data = {
+		cert_file_path = "some/file/some/where"
+		key_file_path = "some/file/some/where/else"
+	}
+}
+`),
+			expectedReturn: 0,
+			expectedStderr: "",
+		},
+		{
+			name: "unknown plugin",
+			config: serverConfigWithPlugins(`
+DataStore "sql" {
+    plugin_data {
+        database_type = "sqlite3"
+        connection_string = "/some/path/to/the/datastore.sqlite3"
+    }
+}
+KeyManager "memory" {
+    plugin_data = {}
+}
+
+UpstreamAuthority "thisdoesnotexist" {
+	plugin_data = {}
+}
+`),
+			expectedReturn: 1,
+			expectedStderr: "Could not validate configuration file: failed to load plugin \"thisdoesnotexist\": no built-in plugin \"thisdoesnotexist\" for type \"UpstreamAuthority\"",
+		},
+		{
+			name: "plugin with bad configuration",
+			config: serverConfigWithPlugins(`
+DataStore "sql" {
+    plugin_data {
+        database_type = "sqlite3"
+        connection_string = "/some/path/to/the/datastore.sqlite3"
+    }
+}
+KeyManager "memory" {
+    plugin_data = {}
+}
+
+UpstreamAuthority "disk" {
+	plugin_data = {
+		cert_file_path = "some/file/some/where"
+	}
+}
```

Comment:
It is a more general ocmment, 
in this approach we are going to depends on a plugin to fail in order for this one to fail, 
it soiunds like we are testing another function but not CLI,
May we fake this to have more control and not be affected by another funcion changes?

Reply:
It's a good point, I'll see what can be done. I don't think we can do that here, but maybe it can be done lower down in the stack.